llvm/lib/Transforms/InstCombine/InstCombineCompares.cpp

```diff
@@ -6855,6 +6855,45 @@ Instruction *InstCombinerImpl::visitICmpInst(ICmpInst &I) {
   if (Value *V = simplifyICmpInst(I.getPredicate(), Op0, Op1, Q))
     return replaceInstUsesWith(I, V);
 
+  {
```

Comment:
This should be inside foldICmpSRemConstant()?

Reply:
@nikic If it's inside `foldIcmpSRemConstant()` the `urem` optimizations won't work.
should I move the `srem` handling, and leave `urem` as it is?

Reply:
You can add a new foldIcmpURemConstant() function next to foldIcmpSRemConstant() (and add it to the switch).